src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs

```diff
@@ -97,13 +100,7 @@ public Endpoint(IInvocationOperation operation, WellKnownTypes wellKnownTypes, S
 
         EmitterContext.HasEndpointParameterMetadataProvider = Parameters.Any(p => p.IsEndpointParameterMetadataProvider);
         EmitterContext.HasEndpointMetadataProvider = Response!.IsEndpointMetadataProvider || Parameters.Any(p => p.IsEndpointMetadataProvider || p.IsEndpointParameterMetadataProvider);
-
-        EmitterContext.HasJsonBodyOrService = Parameters.Any(parameter => parameter.Source == EndpointParameterSource.JsonBodyOrService);
-        EmitterContext.HasJsonBody = Parameters.Any(parameter => parameter.Source == EndpointParameterSource.JsonBody);
-        EmitterContext.HasFormBody = Parameters.Any(parameter => parameter.Source == EndpointParameterSource.FormBody);
-        EmitterContext.HasRouteOrQuery = Parameters.Any(parameter => parameter.Source == EndpointParameterSource.RouteOrQuery);
-        EmitterContext.HasBindAsync = Parameters.Any(parameter => parameter.Source == EndpointParameterSource.BindAsync);
-        EmitterContext.HasParsable = Parameters.Any(parameter => parameter.IsParsable);
+        EmitterContext.HasParsable = Parameters.Any(parameter => parameter.IsParsable || (parameter.Source == EndpointParameterSource.AsParameters && parameter.EndpointParameters.Any(p => p.IsParsable)));
```

Comment:
```suggestion
        EmitterContext.HasParsable ||= Parameters.Any(parameter => parameter.IsParsable || (parameter.Source == EndpointParameterSource.AsParameters && parameter.EndpointParameters.Any(p => p.IsParsable)));
```

Do we really want to set this to `false` if it was already set to `true`?

Reply:
It is not the only place this is assigned.

https://github.com/dotnet/aspnetcore/blob/014e37c7b1efcc457516adc6519738dff15130a0/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs#L188-L192

I like the way we are doing the EmitterContext for the other properties now - we set it to `true` when we know we have one of these things. But here we are doing some sort of "clean up" attempt at the Endpoint level instead of at the EndpointParameter level.

Reply:
Yeah, this has been tracked for a while in https://github.com/dotnet/aspnetcore/issues/47147. I cleaned up some of the setters here but we'll likely want to do a separate pass for emitter context state that is set from the response, for example.